feature/gnoi/system/tests/copying_debug_files_test/README.md

```diff
@@ -6,10 +6,16 @@ Validate that the debug files can be copied out of the DUT.
 
 ## Procedure
 
-*   Issue gnoi System.KillProcessRequest to the DUT to crash a software process. 
-*   Issue gnoi System.Healthz Get RPC to chassis.
-*   TODO: Validate that the device returns the vendor relevant information for
-    debugging via gnoi Healthz.Artifact
+*   Test #1 - Software Process Health Check
+    *   Issue gnoi System.KillProcessRequest to the DUT to crash a software process. 
```

Comment:
I think this test should be rewritten completely

There should only be one function "TestCopyingDebugFiles". Please don't add the new function "TestChassisComponentArtifacts". Idea is to crash a process to create core dumps. Then use healthz.artifacts RPC to collect Core dumps + "Show tech-support or RSI" eqvt.

Also, in place of "SIGNAL_KILL" operation, probably "SIGNAL_ABRT" operation should be used to "terminate the process immediately and dump a core file." as per 
 https://github.com/openconfig/gnoi/blob/main/system/system.proto#L324-L330

Reply:
So based on the offline discussion, the concern here is that the healthz.attributes RPC that checks for different components wont be able to get crash dump for a failed service like the BGP process or RPD process because there doesnt exist corresponding component in OC.

I feel the intent of this test is to ensure we are able to,
1. Collect core-dumps when they happen
2. "healthz artifacts" RPC works for different components.

In order for "1", we can use the gNOI.file.Get operation to collect the core dump. But we need to know the file name of the core dump file. This needs us to use scrape through the show tech output to find the core dump file. Challenge is that the Show tech output will be different for different vendors. I will keep this comment open and lets see if @dplore and @marcushines have some suggestions.

Reply:
So far we only have a requirement  healthz check and artifact for component CHASSIS.  ref:  #1734

Reply:
It is true we expect the CHASSIS artifact to be similar to `show tech support` or other vendor equivalent system wide debug data.  Parsing of the artifact is not required for this test, only confirmation that it is sent and is non-zero bytes.      (ref #1734 )

The use case for this is the healthz artifact for CHASSIS  will be passed to the device developer for root cause analysis.  The device operator is not expected to parse or otherwise directly read this data for this use case.

Reply:
Thanks Darren

Since in "Test #1", killing the process is unrelated to what the subtest is eventually accomplishing with gNOI.Healthz.Getrequest and  gNOI.Healthz.Getresponse on component chassis, it is best to maintain Test#1 given that it provides coverage for gNOI.Healthz.Getrequest and  gNOI.Healthz.Getresponse RPC operations and ensures that the implementation is error free in the NOS being tested. Hence I see value in Test#1. In future as healthz becomes more mature, killing the BGP process and then checking for healthz on the respective component would make most sense. For the timebeing we can maintain test#1 as is.

Test#2 on the other hand (introduced by this pull) with gNOI.healthz.checkrequest and gnoi.healthz.CheckResponse and testing for corresponding Artifacts is accurate.